### PR TITLE
Fix the Table of Contents in README.md

### PR DESCRIPTION
**Pull request by qpakzk**
First, synchronize headers with contents' name.
Equalize uppercase or lowercase letters with headers. Add a question mark.

Second, fix the link of "What is Burrow ?", which didn't work.

Third, change the loction of  "Future work" and "License".
Because the text has a different order with the table of contents.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,14 +9,14 @@ Hyperledger Burrow is a permissioned Ethereum smart-contract blockchain node bui
 
 ## Table of Contents
 
-- [What is burrow](#what-is-burrow)
+- [What is Burrow ?](#what-is-burrow-)
 - [Installation](#installation)
-- [For developers](#for-developers)
+- [For Developers](#for-developers)
```
Comment (silasdavis) [inline README.md]: We are using drop case on other words elsewhere, omit this and I will accept

**Comment by zramsay** [comment]
most if not all of this was fixed on develop in #700

File: README.md (hunk 1 above)
Comment (qpakzk) [reply README.md]: Do you want to undo, otherwise delete "[For developers](#for-developers)"?

Comment (zramsay) [reply README.md]: @qpakzk see my comment below, and PR #700

**Pull request closed without merging**